### PR TITLE
dedicated minimal tests -- `nodes.remove_false_nodes()`

### LINKED ISSUE #60: rough edge in `sgeop.nodes.remove_false_nodes` when multilinestrings passed
When multilinestrings are passed in to `sgeop.nodes.remove_false_nodes()` some strange behavior is observed due to no `.explode()` being called like in the counterpart [`momepy/preprocessing.remove_false_nodes()`](https://github.com/pysal/momepy/blob/f1965cccd48759faf6a2a37739bda310d851124d/momepy/preprocessing.py#L189)

* xref #59

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR:
* adds dedicated minimal tests for `nodes.remove_false_nodes()`
* resolves #60 
* xref 
   * #21 
   * #20
   * https://github.com/pysal/momepy/pull/661
* adds testing
* adds docstrings and type hints

### CODE REVIEW

**Comment by jGaboardi** [comment]
> We should probably also copy all the tests from momepy here.

We are not expecting the same results here, though right? Just trying [the first little block](https://github.com/pysal/momepy/blob/d83197653029256147753b1821b4e4dc2427df04/momepy/tests/test_preprocess.py#L34-L38) locally and it fails.

```
        false_network = geopandas.read_file(momepy.datasets.get_path("tests"), layer="network")
        false_network["vals"] = range(len(false_network))
        fixed = sgeop.nodes.remove_false_nodes(false_network)
>       assert len(fixed) == 56
E       assert 61 == 56
E        +  where 61 = len(                                                geometry  vals\n61.0   LINESTRING (-727482.257 -1052966.07, -727485.5...     0\n62.0   LINESTRING (-727210.943 -1053072.14, -727212.8...     1\n63.0   LINESTRING (-727210.943 -1053072.14, -727206.5...     2\n64.0   LINESTRING (-727141.085 -1053103.182, -727129....     3\n65.0   LINESTRING (-727136.904 -1053091.833, -727117....     4\n...                                                  ...   ...\n117.0  LINESTRING (-727412.897 -1052948.823, -727407....    56\n118.0  LINESTRING (-727248.835 -1052881.77, -727254.5...    57\n119.0  LINESTRING (-727289.896 -1052909.828, -727281....    58\n120.0  LINESTRING (-727227.04 -1052735.582, -727226.1...    59\n121.0  LINESTRING (-727238.493 -1052817.281, -727253....    60\n\n[61 rows x 2 columns])

sgeop/tests/test_nodes.py:399: AssertionError
```

**Comment by martinfleis** [comment]
Not entirely sure on those test data to be fair. The one in sgeop does contain some fixes but it may be worth verifying the differences.

**Comment by jGaboardi** [comment]
> Not entirely sure on those test data to be fair. The one in sgeop does contain some fixes but it may be worth verifying the differences.

OK after the PR on the `momepy` side is settled I will back in this.

**Comment by jGaboardi** [comment]
This seems to specifically be happening in the `sgeop` implementation when passing in a GeoSeries -- passing in a GeoDataFrame succeeds.

**Comment by jGaboardi** [comment]
Key piece [in `momepy.remove_false_nodes()`](https://github.com/pysal/momepy/blob/f1965cccd48759faf6a2a37739bda310d851124d/momepy/preprocessing.py#L187-L189) that is not in `sgeop.remove_false_nodes()`:

```python
# explode to avoid MultiLineStrings
 # reset index due to the bug in GeoPandas explode
 df = gdf.reset_index(drop=True).explode(ignore_index=True)
```

**Comment by jGaboardi** [comment]
When adding [the line above](https://github.com/uscuni/sgeop/pull/59#issuecomment-2433555961) into `sgeop.remove_false_nodes()` all chunks [here](https://github.com/pysal/momepy/blob/f1965cccd48759faf6a2a37739bda310d851124d/momepy/tests/test_preprocess.py#L33-L68) pass (when some index tinkering) ***except for*** the coordinate ordering chunk:

```python
# check loop order
expected = np.array(
    [
        [-727238.49292668, -1052817.28071986],
        [-727253.1752498, -1052827.47329062],
        [-727223.93217677, -1052829.47624082],
        [-727238.49292668, -1052817.28071986],
    ]
)
np.testing.assert_almost_equal(
    np.array(fixed.loc[53].geometry.coords), expected
)
```

This appears to be due to the indices of the resultant edges not lining up, and the index location produced from the `sgeop` implementation is now `55`. Can confirm testing suite passes locally with this adjustment.

File: sgeop/nodes.py [hunk 1]
```diff
@@ -172,25 +172,41 @@ def weld_edges(
...
+
     if len(gdf) < 2:
         return gdf
 
     if isinstance(gdf, gpd.GeoSeries):
         gdf = gdf.to_frame("geometry")
 
+    # explode to avoid MultiLineStrings
+    # reset index due to the bug in GeoPandas explode
+    gdf = gdf.reset_index(drop=True).explode(ignore_index=True)
```
Comment (martinfleis) [inline sgeop/nodes.py:208]:
```suggestion
    gdf = gdf.explode(ignore_index=True)
```

I don't think that workaround is no longer needed. We should also make a note in docstring that mutlilinestrings are exploded.

**Comment by martinfleis** [comment]
Let's also add notes on 3D geometry being downcast, potentially. And about MultiLineStrings being exploded.

**Pull request merged**